pallets/asset/src/lib.rs

```diff
@@ -1502,8 +1637,33 @@ impl<T: Trait> Module<T> {
             .checked_add(&value)
             .ok_or(Error::<T>::BalanceOverflow)?;
 
-        Self::_update_checkpoint(ticker, from_portfolio.did, from_total_balance);
-        Self::_update_checkpoint(ticker, to_portfolio.did, to_total_balance);
+        let is_checkpoint_due = |did| Self::is_checkpoint_due(ticker, did);
+        let maybe_due_checkpoint_from = is_checkpoint_due(from_portfolio.did);
+        let maybe_due_checkpoint_to = if maybe_due_checkpoint_from.is_none() {
+            // Compute the checkpoint only once for either the sender or recipient.
+            is_checkpoint_due(to_portfolio.did)
+        } else {
+            None
+        };
+        if let Some(_) = maybe_due_checkpoint_from
+            .as_ref()
+            .or(maybe_due_checkpoint_to.as_ref())
+        {
+            T::ProtocolFee::charge_fee(ProtocolOp::AssetCheckpoint)?;
```

Comment:
I think we should only be charging the ProtocolOp fee when a schedule is first created, rather than every time it is triggered (which will be by an arbitrary transfer).